### PR TITLE
Routes test for spectrums/set_search added

### LINKED ISSUE #862: Add routes test for spectrums/set_search
Hi, this is a [first-timers-only issue](https://publiclab.github.io/community-toolbox/#r=all). This means we've worked to make it more legible to folks who either **haven't contributed to our codebase before, or even folks who haven't contributed to open source before**. 

If that's you, we're interested in helping you take the first step and can answer questions and help you out as you do. Note that we're especially interested in contributions from people from groups underrepresented in free and open source software!

We know that the process of creating a pull request is the biggest barrier for new contributors. This issue is for you 💝

If you have contributed before, **consider leaving this one for someone new**, and looking through our general [help wanted](https://github.com/publiclab/plots2/labels/help-wanted) issues. Thanks!

### 🤔 What you will need to know.

Nothing. This issue is meant to welcome you to Open Source :) We are happy to walk you through the process.

### 📋 Step by Step

- [ ] 🙋 **Claim this issue**: Comment below. If someone else has claimed it, ask if they've opened a pull request already and if they're stuck -- maybe you can help them solve a problem or move it along!

- [ ] 📝 **Update** the file [routes_test.rb](https://github.com/publiclab/spectral-workbench/blob/main/test/integration/routes_test.rb) in the `spectral-workbench` repository (press the little pen Icon) and edit the line as shown below. 

[See this page](https://publiclab.github.io/community-toolbox/#r=all) for some help in taking your first steps!

Below is a "diff" showing in read (and a `-`) which lines to remove, and in green (and a `+`) which lines to add:

```diff
+  test "test get request for spectrums set_search" do
+    assert_routing({ path: '/spectrums/set_search', method: 'get' }, {controller: 'spectrums', action: 'set_search'})
+  end
```

- [ ] 💾 **Commit** your changes

- [ ] 🔀 **Start a Pull Request**. There are two ways how you can start a pull request:

1. If you are familiar with the terminal or would like to learn it, [here is a great tutorial](https://egghead.io/series/how-to-contribute-to-an-open-source-project-on-github) on how to send a pull request using the terminal.

2. You can also [edit files directly in your browser](https://help.github.com/articles/editing-files-in-your-repository/) and open a pull request from there. 

- [ ] 🏁 **Done** Ask in comments for a review :)

### 🤔❓ Questions?

Leave a comment below!

### Is someone else already working on this?

We encourage you to link to this issue by mentioning the issue # in your pull request, so we can see if someone's already started on it. **If someone seem stuck, offer them some help!** Otherwise, [take a look at some other issues you can help with](https://publiclab.github.io/community-toolbox/#r=all). Thanks!

(This issue was created by [First-Timers-Bot](https://github.com/hoodiehq/first-timers-bot).)

### PR DESCRIPTION
**Pull request by anantraghuvanshi**
Issue #862, required changes are done

Fixes #(update with issue number)

Make sure these boxes are checked before your pull request is ready to be reviewed and merged. Thanks!

* [ ] tests pass -- `rake test`
* [ ] code is in a uniquely-named feature branch and has been rebased on top of the latest master (especially if you've been asked to make additional changes)
* [ ] pull requests are descriptively named
* [ ] if possible, multiple commits squashed if they're smaller changes
* [ ] reviewed/confirmed/tested by another contributor or maintainer

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/wiki/contributing-to-public-lab-software

Thanks!
Fixes #862

### CODE REVIEW

**Comment by anantraghuvanshi** [comment]
Hey @Tlazypanda, I have updated the file after the last test failed, kindly review it!

**Review by Tlazypanda — COMMENTED** [review]
Hey @anantraghuvanshi this actually overwrites one of our already existing tests :sweat_smile: can you please add it as a separate test?

**Comment by anantraghuvanshi** [comment]
> Hey @anantraghuvanshi this actually overwrites one of our already existing tests sweat_smile can you please add it as a separate test?

I get it but how can I do that, may you guide me to that, because i didn't understand what that meant, sorry!

**Review by Tlazypanda — COMMENTED** [review]
Hey @anantraghuvanshi can you check the files changed tab on github where it show the `plot_rss` code being overwritten here? 
So we need to make sure that we are only adding the ` set_search` test without overwriting other existing tests

Once we make this change the pull request can be merged :tada: Thanks :v:

**Comment by anantraghuvanshi** [comment]
Hi @Tlazypanda, this part of the file shows conflict, so if I simply delete that part alone I think it would fix the issue, right?</br> or is it that we need to write only those tests which are not present in `plot_rss`?
image

**Review by Tlazypanda — COMMENTED** [review]
Hey @anantraghuvanshi Have resolved the conflicts but those were other tests not mentioning those if you can just add the `plots_rss` test that is being overwritten here that would be great! :tada: 
Screenshot from 2022-02-08 01-27-07

File: test/integration/routes_test.rb [hunk 1]
```diff
@@ -96,5 +96,9 @@ class RoutesTest < ActionDispatch::IntegrationTest
   test "test get request for spectrums compare_search" do
     assert_routing({ path: '/spectrums/compare_search', method: 'get' }, {controller: 'spectrums', action: 'show', id: 'compare_search'})
   end
+  
+  test "test get request for spectrums set_search" do
+   assert_routing({ path: '/spectrums/set_search', method: 'get' }, {controller: 'spectrums', action: 'show', id: 'set_search'})
```
Comment (TildaDares) [inline test/integration/routes_test.rb:101]:
```suggestion
   assert_routing({ path: '/spectrums/set_search', method: 'get' }, {controller: 'spectrums', action: 'set_search', id: 'set_search'})
```
@anantraghuvanshi The action should be `set_search`

**Review by TildaDares — CHANGES_REQUESTED** [review]

File: test/integration/routes_test.rb (hunk 1 above)
Comment (TildaDares) [reply test/integration/routes_test.rb:101]: Nevermind, the test would fail if the action was changed back to 'set_search'.

**Comment by TildaDares** [comment]
This is ready to be merged!

**Review by TildaDares — APPROVED** [review]
Great work @anantraghuvanshi! Thank you

**Comment by TildaDares** [comment]
Congratulations and welcome to Public Lab @anantraghuvanshi 🎉

**Comment by anantraghuvanshi** [comment]
> Great work @anantraghuvanshi! Thank you

Thanks a lot, @TildaDares and thanks for helping through all the way!

**Pull request merged**